pallets/ethereum-transactions/src/lib.rs

```diff
@@ -454,6 +454,8 @@ impl<T: Config> Pallet<T> {
                 Some(T::ValidatorManagerContractAddress::get()),
             EthTransactionType::ActivateValidator(_) =>
                 Some(T::ValidatorManagerContractAddress::get()),
+            EthTransactionType::ActivateCollator(_) =>
```

Comment:
Maybe we should add a TODO in this function because we only have 1 contract address and most of the match branches return the same one.

Reply:
Can you elaborate please? The above comment also refers to the same line of code.

Reply:
In `get_contract_address`, all match conditions, except publicRoot return the same contract address so we could combine them. Even publishRoot, although it reads from a different storage item, is returning the same contract address so we can add a TODO to simplify the logic here.